lib/source-code/source-code.js

```diff
@@ -411,19 +411,29 @@ class SourceCode extends TokenStore {
     }
 
     /**
-     * Determines if two tokens have at least one whitespace character
-     * between them. This completely disregards comments in making the
-     * determination, so comments count as zero-length substrings.
-     * @param {Token} first The token to check after.
-     * @param {Token} second The token to check before.
-     * @returns {boolean} True if there is only space between tokens, false
-     *  if there is anything other than whitespace between tokens.
+     * Determines if two nodes or tokens have at least one whitespace character
+     * between them.
+     * @param {ASTNode|Token} first The node or token to check after.
+     * @param {ASTNode|Token} second The node or token to check before.
+     * @returns {boolean} True if there is a whitespace character between
+     * any of the tokens found between the two given nodes or tokens.
      * @public
      */
     isSpaceBetweenTokens(first, second) {
-        const text = this.text.slice(first.range[1], second.range[0]);
+        const finalToken = this.getFirstToken(second) || second;
+        let currentToken = this.getLastToken(first) || first;
 
-        return /\s/u.test(text.replace(/\/\*.*?\*\//gus, ""));
+        while (currentToken !== finalToken) {
```

Comment:
What does it happen if I give `first` a token that is after the `second`?

Reply:
That's a good question! That also brings up the question of what to do when a token or node is inside the other node.

Here's what I'd like to propose:

1. If the nodes/tokens are given in reverse order (i.e. the first one occurs after the second), I think we can either throw an error or just perform the check anyway. My preference would be for the latter.
1. For when a token or node is inside a node, I think we should just always return `false`.

Thoughts on this approach?

Reply:
Updated!